File: src/pocketcloud/cloud/network/packet/impl/normal/PlayerKickPacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl\normal;

use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\CloudPacket;
use pocketcloud\cloud\network\packet\data\PacketData;
use pocketcloud\cloud\player\CloudPlayerManager;

class PlayerKickPacket extends CloudPacket {

    public function __construct(
        private string $playerName = "",
        private string $reason = ""
    ) {}

    public function encodePayload(PacketData $packetData): void {
        $packetData->write($this->playerName);
        $packetData->write($this->reason);
    }

    public function decodePayload(PacketData $packetData): void {
        $this->playerName = $packetData->readString();
        $this->reason = $packetData->readString();
    }

    public function getPlayerName(): string {
        return $this->playerName;
    }

    public function getReason(): string {
        return $this->reason;
    }

    public function handle(ServerClient $client): void {
        if (($player = CloudPlayerManager::getInstance()->get($this->playerName)) !== null) $player->kick($this->reason);
    }

    public static function create(string $player, string $reason): self {
        return new self($player, $reason);
    }
}

File: cloud-network/src/main/java/de/pocketcloud/network/packet/impl/PlayerKickPacket.java
package de.pocketcloud.network.packet.impl;

import de.pocketcloud.api.network.packet.AuthenticatedPacket;
import de.pocketcloud.api.network.packet.ClientboundPacket;
import de.pocketcloud.api.network.packet.CloudboundPacket;
import de.pocketcloud.api.network.packet.data.IPacketData;
import de.pocketcloud.network.packet.CloudPacket;
import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@Getter
public final class PlayerKickPacket extends CloudPacket implements ClientboundPacket, CloudboundPacket, AuthenticatedPacket {

    private String player;
    private String reason;
    private String disconnectScreenMessage;

    public PlayerKickPacket(String player, String reason, String disconnectScreenMessage) {
        this.player = player != null ? player : "";
        this.reason = reason != null ? reason : "";
        this.disconnectScreenMessage = disconnectScreenMessage != null ? disconnectScreenMessage : "";
    }

    @Override
    public void encodePayload(IPacketData packetData) {
        packetData.writeAll(player, reason, disconnectScreenMessage);
    }

    @Override
    public void decodePayload(IPacketData packetData) {
        this.player = packetData.readString();
        this.reason = packetData.readString();
        this.disconnectScreenMessage = packetData.readString();
    }

    public static PlayerKickPacket create(String player, String reason, String disconnectScreenMessage) {
        return new PlayerKickPacket(player, reason, disconnectScreenMessage);
    }
}
